packages/data/src/redux-store/index.js

```diff
@@ -333,20 +333,35 @@ function mapResolveSelectors( selectors, store ) {
 			'getCachedResolvers',
 		] ),
 		( selector, selectorName ) => ( ...args ) =>
-			new Promise( ( resolve ) => {
+			new Promise( ( resolve, reject ) => {
 				const hasFinished = () =>
 					selectors.hasFinishedResolution( selectorName, args );
+				const finalize = ( result ) => {
+					const hasFailed = selectors.hasResolutionFailed(
+						selectorName,
+						args
+					);
+					if ( hasFailed ) {
+						const error = selectors.getResolutionFailure(
+							selectorName,
+							args
+						);
```

Comment:
**Suggestion (non-blocking):** I think we can  simplify this by checking if there's an error and rejecting.

```js
const error = selectors.getResolutionFailure(
  selectorName,
  args
);

if ( error ) {
  reject( error );
}
```

Reply:
As much as I'd like that, an error could also be `null` or `undefined`, e.g. consider this snippet:

```js
try {
  throw null;
} catch(e) {
  console.log(e);
}
```